rskj-core/src/main/java/org/ethereum/rpc/Filter.java

```diff
@@ -75,6 +83,9 @@ List<FilterEvent> getEventsInternal() {
     }
 
     protected synchronized void add(FilterEvent evt) {
+        if (maxLogsToReturn > 0 && events.size() + 1 > maxLogsToReturn) {

```

Comment:
probably we should do this in `LogFilter` as we are only limiting logs

Reply:
As it is optional to initialize the limit, I though it was a good place to do this logic. I'll be moving it to LogFilter instead.